tests/models/udop/test_modeling_udop.py

```diff
@@ -0,0 +1,909 @@
+        config,
+    ):
+        prev_vocab_size = config.vocab_size
+
+        config.tie_word_embeddings = False
+        model = UdopUnimodelForConditionalGeneration(config=config).to(torch_device).eval()
+        model.resize_token_embeddings(prev_vocab_size - 10)
+
+        self.parent.assertEqual(model.get_input_embeddings().weight.shape[0], prev_vocab_size - 10)
+        self.parent.assertEqual(model.get_output_embeddings().weight.shape[0], prev_vocab_size - 10)
+        self.parent.assertEqual(model.config.vocab_size, prev_vocab_size - 10)
+
+    def prepare_config_and_inputs_for_common(self):
+        config_and_inputs = self.prepare_config_and_inputs()
+        (
+            config,
+            input_ids,
+            image_input,
+            text_seg_data,
+            image_seg_data,
+            # decoder_input_ids,
+            # attention_mask,
+            # decoder_attention_mask,
+            lm_labels,
+        ) = config_and_inputs
+
+        inputs_dict = {
+            "input_ids": input_ids,
+            # "attention_mask": attention_mask,
+            "image": image_input,
+            "seg_data": text_seg_data,
+            "visual_seg_data": image_seg_data,
+            # "decoder_input_ids": decoder_input_ids,
+            # "decoder_attention_mask": decoder_attention_mask,
+            "labels": lm_labels,
+        }
+        return config, inputs_dict
+
+
+@require_torch
+class UdopModelTest(ModelTesterMixin, unittest.TestCase):
+    all_model_classes = (
+        (UdopUnimodelForConditionalGeneration, UdopDualForConditionalGeneration) if is_torch_available() else ()
+    )
+    all_generative_model_classes = (
+        (UdopUnimodelForConditionalGeneration, UdopDualForConditionalGeneration) if is_torch_available() else ()
+    )
+    all_parallelizable_model_classes = (
+        (UdopUnimodelForConditionalGeneration, UdopDualForConditionalGeneration) if is_torch_available() else ()
+    )
+    fx_compatible = True
+    test_pruning = False
+    test_resize_embeddings = False
+    test_model_parallel = True
+    is_encoder_decoder = True
+    test_attention_outputs = False
+    test_headmasking = False
+    test_inputs_embeds = False
+    test_hidden_states_output = False
+    model_split_percents = [0.8, 0.9]
+
+    def setUp(self):
+        self.model_tester = UDOPModelTester(self)
+        self.config_tester = ConfigTester(self, config_class=UdopConfig, d_model=37)
+
+    def test_config(self):
+        self.config_tester.run_common_tests()
+
+    @unittest.skip("Test has a segmentation fault on torch 1.8.0")
+    def test_export_to_onnx(self):
+        config_and_inputs = self.model_tester.prepare_config_and_inputs()
+        model = UdopUnimodelForConditionalGeneration(config_and_inputs[0]).to(torch_device)
+        with tempfile.TemporaryDirectory() as tmpdirname:
+            torch.onnx.export(
+                model,
+                (config_and_inputs[1], config_and_inputs[3], config_and_inputs[2]),
```

Comment:
@raghavanone Trying to understand how you are exporting the model here. Judging from `prepare_config_and_inputs` you are passing values into the model for (the forward function parameters) `input_ids`, `seg_data` and `image`. But these parameters [are not the first 3 parameters for the forward function](https://github.com/huggingface/transformers/blob/dde1d6c4077e8a8b482de924b86a8324536d1c0e/src/transformers/models/udop/modeling_udop.py#L2368). How does onnx know that you are inputting values for `input_ids`, `seg_data` and `image` in this tuple?

In short, I'm having a lot of problems trying to `torch.jit.trace` or `torch.onnx.export` `UdopUniModelForConditionalGeneration` because (i believe) the forward function parameters I'm passing data into are not at the beginning of the forward function and in consecutive order. If you try to pass in `None` for the parameters you aren't filling you get a `Only Tensors and (possibly nested) Lists, Dicts, and Tuples of Tensors can be traced` error (can't pass in Nonetype).

Reply:
In fact, when I run this test in my env I get:

```
  File "python3.9/site-packages/transformers/models/udop/modeling_udop.py", line 1512, in forward
    bbox = torch.clip(bbox, 0.0, 1.0)
TypeError: clip() received an invalid combination of arguments - got (NoneType, float, float), but expected one of:
 * (Tensor input, Number min, Number max, *, Tensor out)
 * (Tensor input, Tensor min, Tensor max, *, Tensor out)
```
Which is the same error I'm getting while trying to export my finetuned version of it.

Reply:
I am able to get passed this error if I arrange the parameters im filling to the beginning of the forward function and in the same order as passed to torch.onnx.export. However, more errors arise:

```
RuntimeError: 0 INTERNAL ASSERT FAILED at "/Users/runner/work/pytorch/pytorch/pytorch/torch/csrc/jit/ir/alias_analysis.cpp":614, please report a bug to PyTorch. We don't have an op for aten::full_like but it isn't a special case.  Argument types: Tensor, bool, int, int, Device, bool, NoneType,
```

Reply:
@raghavanone you can remove all ONNX export related code from this PR, we can add ONNX support for UDOP in a follow-up PR

Reply:
Also ONNX support now lives fully in optimum, not Transformers.

Reply:
Thanks @NielsRogge, @sgugger, I'll try to not muddy this PR up anymore with unrelated questions, but I'm curious, what would be HuggingFace's process for testing tracing/onnx export of a new model (e.g. UDOP) after the PR supporting the models inclusion into transformers has been merged? Just curious if you can set expectations at all for when users could expect some test code that verifies the UDOP model inside transformers can properly export.